File: Python/1.py
def multiple_of_3_or_5(n):
    for i in xrange(1, n):
        if not (i % 3) or not (i % 5):
            yield i

def main():
    print sum(multiple_of_3_or_5(1000))

if __name__ == "__main__":
    main()

File: Ruby/1.rb
def multiple_of_3_or_5(n)
  sum = 0
  1.upto n-1 do |i|
    if (i % 3 == 0) or (i % 5 == 0)
      sum += i
    end
  end
  sum
end

puts multiple_of_3_or_5(1000)
